### PR TITLE
{vis}[intel/2016a] PCL v1.7.2 (REVIEW)

### PR DESCRIPTION
**Pull request by boegel**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/e/Eigen/Eigen-3.2.8-intel-2016a.eb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+name = 'Eigen'
+version = '3.2.8'
+
+homepage = 'http://eigen.tuxfamily.org/index.php?title=Main_Page'
+description = """Eigen is a C++ template library for linear algebra:
+ matrices, vectors, numerical solvers, and related algorithms."""
+
+toolchain = {'name': 'intel', 'version': '2016a'}
+toolchainopts = {'optarch': True, 'pic': True}
```
Comment (wpoely86) [inline easybuild/easyconfigs/e/Eigen/Eigen-3.2.8-intel-2016a.eb]: euh? Why is this here for a headers only library?

File: easybuild/easyconfigs/p/PCL/PCL-1.7.2-intel-2016a-Python-2.7.11.eb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+builddependencies = [('CMake', '3.5.2')]
+dependencies = [
+    ('Boost', '1.60.0', versionsuffix),
+    ('Eigen', '3.2.8'),
+    ('FLANN', '1.8.4', versionsuffix),
+    ('VTK', '6.3.0', versionsuffix),
+    ('Qhull', '2015.2'),
+]
+
+configopts = '-DCMAKE_BUILD_TYPE=RELEASE '
```
Comment (wpoely86) [inline easybuild/easyconfigs/p/PCL/PCL-1.7.2-intel-2016a-Python-2.7.11.eb]: can be dropped soon :wink:

File: easybuild/easyconfigs/p/PCL/PCL-1.7.2-intel-2016a-Python-2.7.11.eb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+toolchain = {'name': 'intel', 'version': '2016a'}
+toolchainopts = {'pic': True}
+
+source_urls = ['https://github.com/PointCloudLibrary/pcl/archive/']
+sources = [SOURCELOWER_TAR_GZ]
+
+builddependencies = [('CMake', '3.5.2')]
+dependencies = [
+    ('Boost', '1.60.0', versionsuffix),
+    ('Eigen', '3.2.8'),
```
Comment (wpoely86) [inline easybuild/easyconfigs/p/PCL/PCL-1.7.2-intel-2016a-Python-2.7.11.eb]: this is a build dep

File: easybuild/easyconfigs/v/VTK/VTK-6.3.0-intel-2016a-Python-2.7.11.eb [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+source_urls = ['http://www.vtk.org/files/release/%(version_major_minor)s']
+sources = [
+    SOURCE_TAR_GZ,
+    '%(name)sData-%(version)s.tar.gz',
+]
+
+builddependencies = [('CMake', '3.5.2')]
+
+mesaver = '11.0.8'
```
Comment (wpoely86) [inline easybuild/easyconfigs/v/VTK/VTK-6.3.0-intel-2016a-Python-2.7.11.eb]: drop

File: easybuild/easyconfigs/e/Eigen/Eigen-3.2.8-intel-2016a.eb (hunk 1 above)
Comment (boegel) [inline easybuild/easyconfigs/e/Eigen/Eigen-3.2.8-intel-2016a.eb]: hah, good point...

File: easybuild/easyconfigs/p/PCL/PCL-1.7.2-intel-2016a-Python-2.7.11.eb (hunk 3 above)
Comment (boegel) [inline easybuild/easyconfigs/p/PCL/PCL-1.7.2-intel-2016a-Python-2.7.11.eb]: I don't think that's true...

What if the PCL include files have includes on Eigen?

Although I guess in that case both PCL and Eigen should be listed as build deps.

Not sure what the best course is here, and including it as a real dep is low impact?

File: easybuild/easyconfigs/p/PCL/PCL-1.7.2-intel-2016a-Python-2.7.11.eb (hunk 2 above)
Comment (boegel) [inline easybuild/easyconfigs/p/PCL/PCL-1.7.2-intel-2016a-Python-2.7.11.eb]: or not; is there a difference between `Release`, `RELEASE`, etc.?

File: easybuild/easyconfigs/v/VTK/VTK-6.3.0-intel-2016a-Python-2.7.11.eb (hunk 4 above)
Comment (boegel) [inline easybuild/easyconfigs/v/VTK/VTK-6.3.0-intel-2016a-Python-2.7.11.eb]: oops, thx

File: easybuild/easyconfigs/p/PCL/PCL-1.7.2-intel-2016a-Python-2.7.11.eb (hunk 2 above)
Comment (wpoely86) [inline easybuild/easyconfigs/p/PCL/PCL-1.7.2-intel-2016a-Python-2.7.11.eb]: It should be case sensitive I think. Not sure. Check if they do anthing special in their `CMakeLists.txt`?

File: easybuild/easyconfigs/p/PCL/PCL-1.7.2-intel-2016a-Python-2.7.11.eb (hunk 3 above)
Comment (boegel) [inline easybuild/easyconfigs/p/PCL/PCL-1.7.2-intel-2016a-Python-2.7.11.eb]:
```
grep -R 'Eigen' $EASYBUILD_PREFIX/software/PCL/1.7.2-intel-2016a-Python-2.7.11/include | grep '#include' | wc -l
53
```

example:

```
.../software/PCL/1.7.2-intel-2016a-Python-2.7.11/include/pcl-1.7/pcl/point_cloud.h:#include <Eigen/StdVector>
.../software/PCL/1.7.2-intel-2016a-Python-2.7.11/include/pcl-1.7/pcl/point_cloud.h:#include <Eigen/Geometry>
```

@wpoely86: To me, this is a reason to make `Eigen` a runtime dep.

File: easybuild/easyconfigs/p/PCL/PCL-1.7.2-intel-2016a-Python-2.7.11.eb (hunk 2 above)
Comment (boegel) [inline easybuild/easyconfigs/p/PCL/PCL-1.7.2-intel-2016a-Python-2.7.11.eb]: nah, should be `Release`, cfr. http://www.pointclouds.org/documentation/tutorials/compiling_pcl_posix.php#id3

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
node2417.golett.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/2a622f777a88cd758a02d619109ed49c for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
node2159.delcatty.os - Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/be33ded2cadef211c4acd3e89e6859a7 for a full test report.

**Comment by boegel** [comment]
Thanks for the review @wpoely86!
